user-interface/src/main/java/life/qbic/datamanager/views/general/MultiSelectLazyLoadingGrid.java

```diff
@@ -51,16 +51,17 @@ private Checkbox renderCheckbox(T bean) {
 
       if (event.isFromClient()) {
```

Comment:
This selection listener is extremely complex.
Not only is one selection listener added to the `selectAllCheckbox` for every rendered checkbox (including ones that are rendered when scrolling) and never removed. Further the logic is pretty complex and all in one lambda.
The listener on the checkbox should only concern itself with actions related to the selection. There should not be any outside value change listener be added here.
Please refactor this. The functionality of selecting all /none belongs to the `selectAllCheckBox` and not to the individual checkboxes.

Reply:
It's a surprisingly complex problem that took us about a week to implement the last time to get rid of all the edge cases.

* The detach listener is a great idea and works well! 👍 

* Uncoupling "select all" and the selection of the individual boxes does not work so easily, as far as I can tell. The selectAllCheckBox must also change its value when a user selects the last unselected row or when they deselect one row when all others are selected. I don't see an alternative solution that doesn't make the code even more complex.

Reply:
Why does the select all checkbox need to change the value when all rows are selected?

Reply:
following scenario:

1. the user filters for ngs
2. the user selects all
3. the user filters for some property
4. the user has made a type
5. NOTHING is selected anymore (even if the user corrects the typo)

Reply:
But this would be a consequence of the filtering starting while the user is typing. And if I select everything in Google Mail and start the filter (typo or not), nothing is selected anymore, either.
If this is an issue, I suggest we create a new card for a new PR addressing it.

Reply:
I still think this whole select all implementation is more complex than need be. However I will no longer block this PR for that as it only touches on the subject.